pkg/cmd/project/item-edit/item_edit.go

```diff
@@ -0,0 +1,286 @@
+
+# edit an item's number field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --number 1
+
+# edit an item' date field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --date "2023-01-01"
+
+# edit an item's single-select field value
+# you can retrieve the option ID from the output of the 'field-list --format=json' command
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --single-select-option-id OPTION_ID
+
+# edit an item's iteration field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --iteration-id ITERATION_ID
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			t := tableprinter.New(f.IOStreams)
+			config := editItemConfig{
+				tp:     t,
+				client: client,
+				opts:   opts,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runEditItem(config)
+		},
+	}
+
+	editItemCmd.Flags().StringVar(&opts.itemID, "id", "", "ID of the item to edit (required). For draft issues, the ID is for the draft issue content which is prefixed with `DI_`. For other issues, it is the ID of the project item.")
+	editItemCmd.Flags().StringVar(&opts.format, "format", "", "Output format, must be 'json'.")
+
+	editItemCmd.Flags().StringVar(&opts.title, "title", "", "DRAFT ISSUE - Title of the draft issue item to edit.")
+	editItemCmd.Flags().StringVar(&opts.body, "body", "", "DRAFT ISSUE - Body of the draft issue item to edit.")
+
+	editItemCmd.Flags().StringVar(&opts.fieldID, "field-id", "", "ID of the field to update.")
+	editItemCmd.Flags().StringVar(&opts.projectID, "project-id", "", "ID of the project to which the field belongs to.")
+	editItemCmd.Flags().StringVar(&opts.text, "text", "", "Text value to set on the field.")
+	editItemCmd.Flags().Float32Var(&opts.number, "number", 0, "Number value to set on the field.")
+	editItemCmd.Flags().StringVar(&opts.date, "date", "", "The ISO 8601 (YYYY-MM-DD) date value to set on the field.")
+	editItemCmd.Flags().StringVar(&opts.singleSelectOptionID, "single-select-option-id", "", "ID of the single select option value to set on the field.")
+	editItemCmd.Flags().StringVar(&opts.iterationID, "iteration-id", "", "ID of the iteration value to set on the field.")
+
+	editItemCmd.MarkFlagsMutuallyExclusive("text", "number", "date", "single-select-option-id", "iteration-id")
+	_ = editItemCmd.MarkFlagRequired("id")
+
+	return editItemCmd
+}
+
+func runEditItem(config editItemConfig) error {
+	// update draft issue
+	if config.opts.title != "" || config.opts.body != "" {
+		if !strings.HasPrefix(config.opts.itemID, "DI_") {
+			return errors.New("ID must be the ID of the draft issue content which is prefixed with `DI_`")
```

Comment:
Protip: for any errors related to flags and values passed by flags (or arguments), use `cmdutil.FlagErrorf` to construct an error and return that. The resulting error message will be rendered together with the usage synopsis for the command.

Reply:
and also in 38b15a94